### PR TITLE
[UI] Added Database reset modal and FlushMeshSync Button

### LINKED ISSUE #5479: [UI] Add a modal when Reset Database button is clicked on settings page
#### Current Behavior

Currently when we click the `Reset Database` button we don't get any warning or some sort of modal, that confirms our choice. 

#### Desired Behavior

When the button is clicked a modal should be appearing on the UI confirming the user choice of resetting the database.
The modal should describe what the button does and  the modal should have checkboxes that signify different kind of resets that the user wants to perform
 - Reset database completely
 - Just Flush Meshsync
 
 and then two action buttons which close the modal (sort of cancel button )and confirm my choices button.

Screen Shot 2022-05-05 at 19 17 16

#### Implementation

Look for modals implemented on the Meshery UI and try to reuse that.

#### Mockups

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Harshit-Dandriyal**
Signed-off-by: Harshit Dandriyal <56961488+Harshit-Dandriyal@users.noreply.github.com>

**Description**
Added Dialog to Confirm the Reset Database and Added Flush MeshSync Button.

Screenshot from 2022-05-25 19-02-18

Screenshot from 2022-05-25 19-02-20

Screenshot from 2022-05-25 19-02-22

This PR fixes #5479

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by MUzairS15 — CHANGES_REQUESTED** [review]
Please move reset buttons for the database outside `Operator Configuration`.

File: ui/components/MeshConfigComponent.js [hunk 1]
```diff
@@ -879,12 +941,71 @@ handleNATSClick = () => {
...
             color="primary"
             size="large"
-            onClick={() => this.handleReset()}
+            onClick={handleClickOpenDatabaseDialogue}
             className={classes.button}
             data-cy="btnResetDatabase"
           >
               Reset Database
           </Button>
+          <Dialog
```
Comment (Chadha93) [inline ui/components/MeshConfigComponent.js:950]: @Harshit-Dandriyal could you please use this pre-defined Prompt component from Meshery UI, to maintain consistency across modals. https://github.com/meshery/meshery/blob/master/ui/components/PromptComponent.js

**Review by Chadha93 — COMMENTED** [review]
@Harshit-Dandriyal Good Work here, some enhancements suggested.

File: ui/components/MeshConfigComponent.js (hunk 1 above)
Comment (Harshit-Dandriyal) [reply ui/components/MeshConfigComponent.js]: Thanks @Chadha93, Done :+1:

**Comment by Harshit-Dandriyal** [comment]
Screenshot from 2022-06-06 01-38-04

@MUzairS15 Our reset buttons for the database are outside of Operator Configuration here :smile: .

File: ui/components/MeshConfigComponent.js [hunk 2]
```diff
@@ -553,63 +552,82 @@ handleNATSClick = () => {
...
+        this.props.updateProgress({ showProgress : true });
+        const self = this;
+        resetDatabase({
+          selector : {
+            clearDB : "true",
+            ReSync : "true", // True - For Hard reset, False otherwise
+            hardReset : "true",
+          },
+          // For now by default set to hard reset.
+          // TODO: User should be able to select soft or hard reset when "Reset Daatabase" btn is clicked.
```
Comment (MUzairS15) [inline ui/components/MeshConfigComponent.js:572]: Can you please remove these comments?? No longer needed.

**Review by MUzairS15 — APPROVED** [review]
LGTM 👍

**Comment by MUzairS15** [comment]
// @Chadha93

**Comment by MUzairS15** [comment]
@Harshit-Dandriyal Will you please refactor, these modals in new [Settings](https://github.com/meshery/meshery/pull/5568) page.
Or else I will do it.

File: ui/components/MeshConfigComponent.js (hunk 2 above)
Comment (Harshit-Dandriyal) [reply ui/components/MeshConfigComponent.js:572]: ok :+1:

**Comment by Harshit-Dandriyal** [comment]
> @Harshit-Dandriyal Will you please refactor, these modals in new [Settings](https://github.com/meshery/meshery/pull/5568) page. Or else I will do it.

I can't directly commit to that pull request, should I wait for that pull request to be merged?

**Review by leecalcote — APPROVED** [review]
This section will soon change.

**Comment by leecalcote** [comment]
@MUzairS15 note the change in language used. The buttons here are not using the theme standard.

**Pull request merged**